openwisp_radius/api/views.py

```diff
@@ -294,16 +295,23 @@ def get_user(self, serializer, *args, **kwargs):
 
     def validate_membership(self, user):
         if not (user.is_superuser or user.is_member(self.organization)):
-            if OrganizationUser.objects.filter(user=user).exists():
+            try:
+                org_registration_enabled = (
+                    self.organization.radius_settings.get_registration_enabled()
+                )
+            except OrganizationRadiusSettings.DoesNotExist:
+                org_registration_enabled = app_settings.REGISTRATION_API_ENABLED
+            if org_registration_enabled:
                 OrganizationUser.objects.create(
                     user=user, organization=self.organization
                 )
             else:
                 message = _(
-                    'The user {username} is not member of any organization.'
-                ).format(username=user.username)
-                logger.warning(message)
-                raise serializers.ValidationError({'non_field_errors': [message]})
+                    'Registration is disabled for {organization} organization'.format(
```

Comment:
Please change this message to `{organization} does not allow self registration of new accounts.`